src/makeCalculator.js

```diff
@@ -5,6 +5,45 @@
  */
 function makeCalculator() {
   // write code here
+  const calculator = {
+    result: 0,
+
+    add(value) {
+      this.result += value;
+
+      return this;
+    },
+
+    subtract(value) {
+      this.result -= value;
+
+      return this;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this;
+    },
+
+    divide(value) {
+      this.result /= value;
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(operation, value) {
+      return operation.call(this, value);
```

Comment:
The `operate` method should not return the result of the operation directly. Instead, it should set the appropriate value to the `result` property and then return the calculator object to allow method chaining. You can achieve this by calling the operation on `this.result` and then returning `this`. Also, consider validating that the `operation` parameter is a function before attempting to call it.